core/hydromet_stratified.py

```diff
@@ -384,6 +384,201 @@ def Scenarios_low_and_high_S(df_runoff_SR1: pd.DataFrame, mu: float, GEV_paramet
+    ratio_to_24h.at[118, 'ratio'] = df2.at[11.75, 'ratio']+0.4*(df2.at[11.875,'ratio']-df2.at[11.75, 'ratio'])
+    ratio_to_24h.at[119, 'ratio'] = df2.at[11.875, 'ratio']+0.6*(df2.at[11.917,'ratio']-df2.at[11.875, 'ratio'])
+    ratio_to_24h.loc[121:240, 'ratio'] = 1-ratio_to_24h.loc[0:119, 'ratio'].sort_index(ascending=False).values
+    ratio_to_24h.loc[120, 'ratio'] = ratio_to_24h.at[121, 'ratio']-(df.at['05m', 'ratio']+1.0/5.0*
+                                                                   (df.at['10m','ratio']-df.at['05m','ratio']))
+    ratio_to_24h.loc[0, 'ratio'] = 0
+    ratio_to_24h['slope'] = ratio_to_24h['ratio'].diff()/0.1
+    ratio_to_24h.at[0, 'slope'] = 0
+    ratio_to_24h['t_step'] = ratio_to_24h.index*0.1
+    ratio_to_24h.index = ratio_to_24h.index*0.1
+    return ratio_to_24h
+
+
+def get_hyeto_input_data_nrcs(temporal_precip_table_dir: str, event: any,
+                         display_print: bool=True) -> pd.DataFrame:
+    '''Extracts the temporal distribution from precipitation frequency data for the specified duration from an Excel 
+       sheet and returns the data as a dataframe. 
+    '''
+    hyeto_precip = 'nrcs_hye_{}'.format(event)
+    df = pd.read_excel(temporal_precip_table_dir, sheet_name=hyeto_precip, index_col=0)
+    if display_print: 
+        print(display(df.head(2)))
+    return df
+
+def get_hyeto_input_data_atlas(temporal_precip_table_dir: str, quartile: any,
+                         display_print: bool=True) -> pd.DataFrame:
+    '''Extracts the temporal distribution from precipitation frequency data for the specified duration from an Excel 
+       sheet and returns the data as a dataframe. 
+    '''
+    hyeto_precip = 'atlas_hye_{}'.format(quartile)
+    df = pd.read_excel(temporal_precip_table_dir, sheet_name=hyeto_precip, index_col=0)
+    weights_df = pd.read_excel(temporal_precip_table_dir, sheet_name='atlas_hye_weights', index_col=0)
+    if display_print: 
+        print(display(df.head(2)))
+    return df, weights_df
+    
+def hydro_out_to_dic(curve_df: pd.DataFrame, BCN: str) -> dict:
+    '''This function takes the dataframe and adds additional data required for the dss file and json file creation.
+    '''
+    dic = {}
+    df_dic = curve_df.to_dict()
+    dates = list(curve_df.index)
+    ordin = curve_df.index.name.title()
+    events = {}
+    for k, v in df_dic.items():
+        if 'E' in k:
+            events[k] = list(v.values())
+    key = 'H24'
+    val = {'time_idx_ordinate': ordin, 
+            'run_duration_days': str(2),
+            'time_idx': dates, 
+            'pluvial_BC_units': 'inch/ts', 
+            'BCName': {BCN: events}}         
+    dic[key] = val
+    return dic
+
+
+def Rename_Final_Events_Precip_Stratified(curve_weight: dict, hydrology: int) -> dict:
+    '''Creates a unique event name based on the duration and relative recurrence interval.
```

Comment:
This description should be updated

Reply:
Now reads: Creates a unique event name based on the recurrence interval, infiltration condition, and temporal distribution.